### PR TITLE
fix: Handle model name changes [DET-6648]

### PR DESCRIPTION
**Pull request by mapmeld**
## Description

When a model successfully changes name, the browser should be redirected to the new URL (/det/models/:new_name)

When a model fails to change name, we were already resetting InlineEditor to the previous name, but this should handle it with a returned error instead of my previous `throw Error()` idea

## Test Plan

1. Change the name of a model "test" -> "test2"
2. Change the name of a model to something not allowed "hello" -> "hello/world"

## Checklist

- [x] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
- [x] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: webui/react/src/pages/ModelDetails.tsx [hunk 1]
```diff
@@ -292,14 +293,16 @@ const ModelDetails: React.FC = () => {
         body: { name: editedName },
         modelName,
       });
+      window.location.href = `/det${paths.modelDetails(editedName)}`;
```
Comment (EmilyBonar) [inline webui/react/src/pages/ModelDetails.tsx:296]: This line doesn't work (at least on Netlify). You should be able to do `routeToReactUrl(paths.modelDetails(editedName))` instead.

File: webui/react/src/pages/ModelDetails.tsx [hunk 2]
```diff
@@ -292,14 +293,16 @@ const ModelDetails: React.FC = () => {
...
     } catch (e) {
       handleError(e, {
         publicSubject: 'Unable to save name.',
         silent: true,
         type: ErrorType.Api,
       });
-      throw Error('Revert name change');
+      return e;
     }
+    return null;
```
Comment (EmilyBonar) [inline webui/react/src/pages/ModelDetails.tsx:305]: Is this `return null` necessary?

File: webui/react/src/components/InlineEditor.tsx [hunk 3]
```diff
@@ -62,14 +62,12 @@ const InlineEditor: React.FC<Props> = ({
 
   const save = useCallback(async (newValue: string) => {
     if (onSave) {
-      try {
-        setIsSaving(true);
-        await onSave(newValue);
-      } catch (e) {
+      setIsSaving(true);
+      const err = await onSave(newValue);
+      if (err !== null) {
```
Comment (EmilyBonar) [inline webui/react/src/components/InlineEditor.tsx:67]: You should be able to just use `if (err)`

**Review by EmilyBonar — CHANGES_REQUESTED** [review]
Looks good, but the redirect needs to work

File: webui/react/src/components/InlineEditor.tsx (hunk 3 above)
Comment (mapmeld) [reply webui/react/src/components/InlineEditor.tsx:67]: I get this TypeScript error `An expression of type 'void' cannot be tested for truthiness.  TS1345`
I'm making the other changes but `onSave` has to have a consistent Promise<object> type or keep this if err !== null

**Review by EmilyBonar — APPROVED** [review]
LGTM! The only comment I have is that it fails silently, should there be a message telling users that updating the name didn't work?

**Comment by mapmeld** [comment]
I agree that there could be a message (above or below the input? using the planned universal notifications button?).
I'd also like to use catch save errors like this in other places where we use InlineEditor.
I'm going to make that into a new ticket.

**Pull request merged**